python/pyarrow/_compute.pyx

```diff
@@ -1933,6 +1933,34 @@ class PartitionNthOptions(_PartitionNthOptions):
         self._set_options(pivot, null_placement)
 
 
+cdef class _CumulativeSumOptions(FunctionOptions):
```

Comment:
Could we delete this cdef class and just inherit from `_CumulativeOptions`?

Reply:
Sure, Have deleted. 
But if we do that, the error message will report  `"`start` type for CumulativeOptions"` for CumulativeSumOptions.

Reply:
Ah good point. I think that's okay, but I am also okay with duplicating it if you prefer that instead.

Reply:
Ok. Let's keep the deteled version